util/random/rand.go

```diff
@@ -76,3 +81,51 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 func New() cipher.Stream {
 	return &randstream{}
 }
+
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
+
+type mixedrandstream struct {
+	Readers []io.Reader
+}
+
+func (r *mixedrandstream) XORKeyStream(dst, src []byte) {
+
+	l := len(dst)
+	if len(src) != l {
+		panic("XORKeyStream: mismatched buffer lengths")
+	}
+
+	var b bytes.Buffer
+	var nerr int
+	for _, reader := range r.Readers {
+		buff := make([]byte, READER_BYTES)
+		n, err := io.ReadFull(reader, buff)
+		if err != nil {
+			nerr++
+		}
+		b.Write(buff[:n])
+	}
+
+	// we are ok with few sources being insecure (i.e., providing less than
+	// READER_BYTES bytes), but not all of them
+	if nerr == len(r.Readers) {
+		panic("all readers failed")
+	}
+
+	// create the XOF output, with hash of collected data as seed
+	h := sha512.New()
```

Comment:
Can you create a variable in this package with a clear name related to mixed random stream to indicate the choice of the hash function ?